### ISSUE TITLE
Cannot get this to work

### ISSUE THREAD

**Issue by DasKatzchen** [bitcraft/dcdrop#1]
It always throws an "ImportError: No module named Upload"

In addition, attempting to run it on python 2.5 as recommended complains that "with" will become a reserved word in 2.6 leading me to wonder if you really mean 2.5 or a later version. I've tried it with multiple versions of python and none seem to have the upload module.

**Comment by bitcraft** [comment]
It's probably been over 7 years since I've used this.  I'll look into it this week.  From just a very brief look, it might be a missing file from the repo....so easy fix.

Just curious, how is your DC connected to the PC....  modem?

**Comment by DasKatzchen** [comment]
It will be connected by modem, specifically using DreamPi, as soon as my modem actually arrives. I picked the first modem on amazon that said it worked in Linux, but turns out I bought it from China and it won't arrive until at _least_ the 4th.

**Comment by bitcraft** [comment]
Since I don't have a setup for testing this anymore, do you mind testing the current HTTP server as well as an updated one?  I'll let you know more after your modem arrives (and if you agree to help out).

**Comment by DasKatzchen** [comment]
I'm down. I went so far as to make a virtualbox instance of Debian Lenny when I was trying to make it work

**Comment by bitcraft** [comment]
Wow, that's dedication.  I've actually got an emulator going to test the PlanetWeb browser and maybe test it all out without a hardware setup.  If its all good emulated, I won't need you to test it.  :smile:  

What python is on the DreamPi, 2.7?  Maybe this could be added to DreamPi and autorun on startup?

**Comment by DasKatzchen** [comment]
The only problem with running it on the DreamPi itself is getting the files off the DreamPi. I'm not sure what python is on it, but putting a different one on is relatively easy.

**Comment by bitcraft** [comment]
Yeah, I hear you about getting files off the pi.  I use ssh/scp all the time, but I know not everybody is like that.  Opening a FTP server on the DreamPi would make file transfer easier.

I'm open to any suggestions.  I've got my DC sitting next to me actually, but my modem is PCI and I don't have a computer with a PCI slot anymore.  :(  I feel like breaking this out this winter.  :D

I'll just target python 2.7 and 3.4, since they are common.

**Comment by DasKatzchen** [comment]
You mean you don't keep an old windows XP box with pci slots, parallel and serial ports, and a floppy drive? How do you connect all your old 90s hardware? :-) 

Yeah, 2.7 and 3.4 seem like a reasonable target.

**Comment by bitcraft** [comment]
You laugh, but I found that my secondary computer actually has a single PCI slot!  Now I need to dig that modem out of storage.  Its an old school hardware one.  My fallback is an external 56k modem that I've lost the cables for.

My emulator experiment failed though.  So I will need to do hardware tests.  Fun times....

**Comment by DasKatzchen** [comment]
Oh I wasn't laughing. I was actually describing a computer on my workbench. It also dual boots into Windows 98 for when XP is too newfangled.

**Comment by DieKatzchen** [comment]
Hey, quick note, you left your .txt file full of product keys in the dev branch. Might want to remove that. Here's how to delete it: https://help.github.com/articles/remove-sensitive-data/

**Comment by bitcraft** [comment]
wow, i cannot believe that happened!  thanks for the heads up, i'll get that clean up right away.

**Comment by bitcraft** [comment]
Just an update.  I'm going to look into this project this weekend...  I got my modem and dreamcast all together again.

**Comment by DasKatzchen** [comment]
Hey, if you're still working on this, finally got my dreamcast connected to wifi. Any progress?

**Comment by bitcraft** [comment]
Wifi?  How did you do that?  Sounds like fun.  Several months ago I think that I fixed all the bitrot.  I guess I haven't uploaded anything though.  I will look at it this weekend.

**Comment by DasKatzchen** [comment]
With a pi 3. The tricky part is that the particular version of Raspbian that runs all the scripts doesn't have an easy way to hook it up to Wi-Fi. You have to add a keyboard and monitor. The version in the dev branch runs, but times out when uploading. Don't know if droppy2.py will help, on the dreamcast the file choose button fails to appear, so I can't try.

**Comment by bitcraft** [comment]
I had to heavily modify the droopy 1 script for the DC browser to work with it.  I have started doing the same with droopy 2.

**Comment by DasKatzchen** [comment]
Awesome. Looking forward to it.

**Comment by allycatty** [comment]
Anybody who's been crazy enough to want to do this in 2023 can use my fork that works out of the box on a dreampi. Also made a script to get stuff back onto the VMU.
https://github.com/allycatty/dcdrop
